container/libcontainer/handler.go

```diff
@@ -758,6 +756,85 @@ func (h *Handler) GetProcesses() ([]int, error) {
 	return pids, nil
 }
 
+// ResetWss to be run as gorutine for non blocking referenced bytes reset
+func (h *Handler) ResetWss(containerName string) error {
+	var err error = nil
+	if *referencedResetInterval == time.Duration(0) {
+		return err
+	}
+	resetInterval := *referencedResetInterval
+	time.Sleep(time.Duration(rand.Intn(int(resetInterval.Seconds()))))
+
+	for {
+		select {
+		case <-h.ReferencedResetStopper:
+			klog.V(5).Infof("Finished reseting wss for %s", containerName)
+			return err
+
+		default:
+			pids, err := h.cgroupManager.GetPids()
+			if err != nil {
+				klog.V(5).Infof("Reset failed to collect pids for handler %s", containerName)
+
+			}
+			if len(pids) == 0 && err == nil {
+				klog.V(5).Infof("Tried to reset wss for container without PIDs %s", containerName)
+
+			}
+			start := time.Now()
+			h.referencedMemoryMutex.Lock()
+			clearReferencedBytes(pids)
+			h.referencedMemoryMutex.Unlock()
+			elapsed := time.Since(start)
+			if elapsed > *referencedResetInterval {
+				klog.Warningf("Exceeded referenced reset interval for %s", containerName)
+				elapsed = elapsed % *referencedResetInterval
+			}
+			toSleep := *referencedResetInterval - elapsed
+			time.Sleep(toSleep)
+		}
+	}
+}
+
+// ReadSmaps to be run as gorutine for non blocking referenced bytes read
+func (h *Handler) ReadSmaps(containerName string) error {
+	var err error = nil
+	klog.Infof("Starting WSS collection for %s", containerName)
```

Comment:
Log level for this information should be adjusted to not overload user with huge amount of logs. Please test it and adjust.